File: Samples/get_set_formulas_results_csharp_vs2008/get_set_formulas_results_csharp_vs2008/page_setresults.cs
using IVisio = Microsoft.Office.Interop.Visio;

public static partial class CSharpSamples
{
    public static void Page_SetResults(IVisio.Document doc)
    {
        var pages = doc.Pages;
        var page = pages.Add();
        page.NameU = "PSR";

        var shape = page.DrawRectangle(1, 1, 4, 3);
        shape.get_CellsU("Width").Formula = "=(1.0+2.5)";
        shape.get_CellsU("Height").Formula = "=(0.0+1.5)";

        // BUILD UP THE REQUEST
        short flags = 0;
        var items = new[]
        {
            new {   shapeid = (short) shape.ID,
                    section = (short) IVisio.VisSectionIndices.visSectionObject, 
                    row     = (short) IVisio.VisRowIndices.visRowXFormOut, 
                    cell    = (short) IVisio.VisCellIndices.visXFormWidth,
                    result = 8.0 ,
                    unitcode = (short) IVisio.VisUnitCodes.visNoCast },    

            new {   shapeid = (short) shape.ID,
                    section = (short) IVisio.VisSectionIndices.visSectionObject, 
                    row     = (short) IVisio.VisRowIndices.visRowXFormOut, 
                    cell    = (short) IVisio.VisCellIndices.visXFormHeight,
                    result = 1.0 ,
                    unitcode = (short) IVisio.VisUnitCodes.visNoCast }
        };

        // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        short[] SID_SRCStream = new short[items.Length * 4];
        object[] results_objects = new object[items.Length];
        object[] unitcodes = new object[items.Length];
        for (int i = 0; i < items.Length; i++)
        {
            SID_SRCStream[i * 4 + 0] = items[i].shapeid;
            SID_SRCStream[i * 4 + 1] = items[i].section;
            SID_SRCStream[i * 4 + 2] = items[i].row;
            SID_SRCStream[i * 4 + 3] = items[i].cell;
            results_objects[i] = items[i].result;
            unitcodes[i] = items[i].unitcode;
        }

        // EXECUTE THE REQUEST
		System.Array SID_SRCStream_sa = (System.Array) SID_SRCStream;
		System.Array results_objects_sa = (System.Array) results_objects;
		System.Array unitcodes_sa = (System.Array) unitcodes;
        int count = page.SetResults(ref SID_SRCStream_sa, ref unitcodes_sa, ref results_objects_sa, flags);

        shape.Text = "SetResults";
    }
}

File: Samples/visio_managed_code_interop_vs2010/VS2010_IronPython_Samples/Page_SetResults.py
import clr 
import System 

clr.AddReference("Microsoft.Office.Interop.Visio") 
import Microsoft.Office.Interop.Visio 
IVisio = Microsoft.Office.Interop.Visio 

from Records import *
import Util


def Page_SetResults( doc ):

        pages = doc.Pages
        page = pages.Add()
        page.NameU = "PSR"
        
        shape = page.DrawRectangle(1, 1, 4, 3)
        shape.CellsU["Width"].Formula = "=(1.0+2.5)"
        shape.CellsU["Height"].Formula = "=(0.0+1.5)"

        # BUILD UP THE REQUEST
        flags = System.Int16(0)        
        items = [
                Page_SetResults_Record( shape.ID, IVisio.VisSectionIndices.visSectionObject, IVisio.VisRowIndices.visRowXFormOut, IVisio.VisCellIndices.visXFormWidth, 8.0, IVisio.VisUnitCodes.visNoCast),                
                Page_SetResults_Record( shape.ID, IVisio.VisSectionIndices.visSectionObject, IVisio.VisRowIndices.visRowXFormOut, IVisio.VisCellIndices.visXFormHeight, 1.0, IVisio.VisUnitCodes.visNoCast)
        ]

        # MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        SRCStream = Util.get_new_system_array(System.Int16, len(items)*4)
        results = Util.get_new_system_array(System.Object, len(items))
        unitcodes = Util.get_new_system_array(System.Object, len(items))
        for i in xrange(len(items)) :
                SRCStream[i * 4 + 0] = items[i].ShapeID
                SRCStream[i * 4 + 1] = items[i].SectionIndex
                SRCStream[i * 4 + 2] = items[i].RowIndex
                SRCStream[i * 4 + 3] = items[i].CellIndex
                results[i] = items[i].Result
                unitcodes[i] = items[i].UnitCode

        # EXECUTE THE REQUEST
        results_sa = Util.get_ref_to_system_array(System.Object,results)
        unitcodes_sa = Util.get_ref_to_system_array(System.Object,unitcodes)
        SRCStream_sa = Util.get_ref_to_system_array(System.Int16,SRCStream)  
        count = page.SetResults(SRCStream_sa, unitcodes_sa, results_sa, flags)

        shape.Text = System.String.Format("SetResults")

